### DISCUSSION TITLE
MLAG: Need to introduce 'virtual' nodes?

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1543] [Features]
Working on MLAG support, I'm thinking we may need the ability to create virtual nodes to represent the "M" side of MLAG link groups, similar to what we do for links in the vlan module.

The ```lag``` module would create a virtual node for the M-side, mark it as ```role: virtual``` to avoid allocation of loopback interfaces, mgmt IPs and the like. Then, after transformation, it would copy the interfaces from the virtual nodes into their 'real' counterparts (groups of mlag peers, typically pairs) and remove the virtual nodes from the final topology

Thoughts?

**Comment by ddutt** [comment]
is your reason for doing this to avoid misconfigs?

**Reply by jbemmel** [reply]
I’m thinking it may be a natural way to present the pair as one logical device to the rest of the code

**Reply by ddutt** [reply]
Hmm, not so sure till I see more. More importantly will this prevent me from misconfguring deliberately? I modeled a bit of MLAG for my playbooks, I don't recall needing to invent a virtual node. Hard to say more about your proposal without knowing a bit more.

Thank you for doing this work.

**Comment by ipspace** [comment]
I don't see why we would need that.

Networking industry commonly used two ways to implement MLAG -- shared control plane and independent devices. If you're thinking about a shared control plane implementation, then you'll have to solve a bunch of other issues as well. For independent devices we don't need a virtual node.

So, as @ddutt said, we'd need to know more about your ideas and what problem you're trying to solve.

**Reply by jbemmel** [reply]
Thanks for the feedback, I have abandoned the idea. PR is ready for review
